analyses/molecular-subtyping-EPN/02_ependymoma_generate_all_data.py

```diff
@@ -176,9 +177,20 @@ def fill_df_with_fpkm_zscores(df, fpkmdf, gene_name):
 
 # Adding Z-scores to dataframe
 expression_cols = ["RELA", "L1CAM", "ARL4D", "CLDN1", "CXorf67", "TKTL1", "GPBP1", "IFT46"]
-for gene in expression_cols:
-    EPN_notebook = fill_df_with_fpkm_zscores(EPN_notebook, fpkm_df, gene)
+# samples with rna matched with dna
+EPN_notebook_rna_dna = EPN_notebook[EPN_notebook["Kids_First_Biospecimen_ID_RNA"].notna()]
+# samples with only dna
+EPN_notebook_dna_only = EPN_notebook[EPN_notebook["Kids_First_Biospecimen_ID_RNA"].isna()]
 
+for gene in expression_cols:
+    # rna matched dna samples
+    EPN_notebook_rna_dna = fill_df_with_fpkm_zscores(EPN_notebook_rna_dna, fpkm_df, gene)
```

Comment:
`EPN_notebook_rna_dna` has already been filtered for NA values on line 181, but then `fill_df_with_fpkm_zscores()` checks again? Do we need the changes to `fill_df_with_fpkm_zscores()` for usage elsewhere? Or can we eliminate the splitting and recombining here and do it all within `fill_df_with_fpkm_zscores()`?

Reply:
Updated fill_df_with_fpkm_zscores() to add NA to WGS only samples and zscore to RNAseq samples instead of splitting and recombining. 
However the issue with possible change in order from your above comment remains to be worked on.